config/seed-peer-generator/src/main.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use libra_config::config::{PersistableConfig, HANDSHAKE_VERSION};
+use libra_network_address::NetworkAddress;
+use libra_secure_json_rpc::JsonRpcClient;
+use libra_types::{on_chain_config::ValidatorSet, validator_info::ValidatorInfo, PeerId};
+use std::{collections::HashMap, convert::TryFrom, path::PathBuf};
+use structopt::StructOpt;
+use libra_types::account_address::AccountAddress;
+
+// TODO: Use the definition from network?
+pub type SeedPeersConfig = HashMap<PeerId, Vec<NetworkAddress>>;
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Debug, StructOpt)]
+#[structopt(about = "Tool to generate configs from chain")]
+enum Args {
+    #[structopt(about = "Generate seed peers config")]
+    GenerateSeedPeers(GenerateArgs),
```

Comment:
Why not inline args below? Are we doing to use this for other things? The only other operation I could imagine is to update an existing config, but I don't know if we need this unflattened layout.

Reply:
Ah, I just was doing something quick and dirty by copying the pattern made by ConfigGenerator.  I'm sure I can flatten it.

I wonder if we should just put it in the `ConfigGenerator`